src/main/scala/akka/di/example/App.scala

```diff
@@ -0,0 +1,20 @@
+package akka.di.example
+
+import akka.actor.Actor
+import akka.di.example.AppSupervisor.Start
+import akka.di.guice.ActorDep
+import com.google.inject.Inject
+
+object AppSupervisor {
+  object Start
+}
+
+class AppSupervisor @Inject() (printer: ActorDep[PrinterActor]) extends Actor {
```

Comment:
Trasformare l'entry point `App` in un actor ha qualche vantaggio pratico o e' solo per eleganza? Immagino che diventa piu' facile gestire logiche di "supervisor" visto il nome ;-) ...

Reply:
Chiamiamola eleganza, l'entry point di un actor system è un attore, la responsabilità della supervisione di un actor system nella sua interezza è del guardiano di livello più alto.

per me gli attori sono una piramide, con un vertice, le piramidi tronche non mi piacciono poichè al primo livello ci sono dei peer che anzichè scalare su un vertice si coordinano fra loro quindi si conoscono

n attori top level si coordinano con n attori top level

1 attore top level si coordina con n attori second level

mi sembra più gestibile

avere un supervisor ed un entrypoint come ActorRef esorta questo tipo di ragionamento